AppSettingsManagerAPI/AppSettingsManagerApi.Infrastructure/MySql/PermissionsManager.cs

```diff
@@ -0,0 +1,75 @@
+// The PermissionsManager class implements all the methods needed to 
+// check whether the user has the necessary permissions to access a settingGroup
+// and allows the user to request permission to a settingGroup if already don't have it
+// The user (owner of the project or admin) can then approve or reject the request
+public class PermissionsManager
+{
+    private readonly List<Permission> _permissions;
+
+    public PermissionsManager()
+    {
+        _permissions = new List<Permission>();
+    }
+
+    public void AddPermission(Permission permission)
+    {
+        _permissions.Add(permission);
+    }
+
+    public bool CanUserAccessSetting(User user, SettingGroup settingGroup, PermissionLevel requiredPermissionLevel)
+    {
+        var permission = _permissions.FirstOrDefault(p => p.UserId == user.Id && p.SettingGroupId == settingGroup.SettingGroupId);
+
+        if (permission == null)
+        {
+            return false;
+        }
+
+        if (permission.CurrentPermissionLevel >= requiredPermissionLevel)
+        {
+            return true;
+        }
+
+        return false;
+    }
```

Comment:
```suggestion
    public async Task<bool> CanUserAccessSetting(string userId, string settingGroupId, PermissionLevel requiredPermissionLevel)
    {
        var settingGroup = await _settingsContext.SettingGroups.Include(s => 
            s.Permissions).SingleAsync(s => s.Id == settingGroupId);
        
        var permission = settingGroup.Permissions.SingleOrDefault(p => p.UserId == userId);
        
        if (permission.CurrentPermissionLevel >= requiredPermissionLevel)
        {
            return true;
        }

        return false;
    }
```

Should make the method async (which also means wrapping the return in a Task<>). Don't need to receive full User and SettingGroup objects.

Also, probably useful to note, if you're retrieving something from the database and you also want data from another table it's linked to. You have to use that .Include() method. So in this case, we're getting a SettingGroup but we want the Permissions it's related to as well so we .Include(s => s.Permissions).